votor/src/certificate_pool.rs

```diff
@@ -467,18 +446,18 @@ impl CertificatePool {
         });
         let vote_type = VoteType::get_type(vote);
         if let Some(conflicting_type) =
-            self.has_conflicting_vote(slot, vote_type, &validator_vote_key, &block_id)
+            self.has_conflicting_vote(vote_slot, vote_type, &validator_vote_key, &block_id)
```

Comment:
this should not to be part of this PR, but why do we turn the `rank` into the vote key and then use the vote key to index validators? I think we should only use the rank within all of Pool